Source/Core/Quaternion.js

```diff
@@ -184,17 +188,29 @@ define([
      * @param {Quaternion} [result] The object onto which to store the result.
      * @returns {Quaternion} The modified result parameter or a new Quaternion instance if none was provided.
      */
-    Quaternion.fromHeadingPitchRoll = function(heading, pitch, roll, result) {
+    Quaternion.fromHeadingPitchRoll = function(headingOrHeadingPitchRoll, pitchOrResult, roll, result) {
```

Comment:
I don't see how to use options parameter in Quaternion.fromHeadingPitchRoll as the first parameter will be a HeadingPitchRoll object?
You suggest that rollQuaternion, pitchQuaternion and headingQuaternion become scratch variables? Ok for me but just for information that was the previous design.